core/src/main/java/org/eclipse/hono/util/TenantConstants.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.util;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import io.vertx.core.json.DecodeException;
+import org.apache.qpid.proton.message.Message;
+
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Constants &amp; utility methods used throughout the Tenant API.
+ */
+
+public final class TenantConstants extends RequestResponseApiConstants {
+
+    /* tenant actions */
+    public static final String ACTION_GET                        = "get";
+    public static final String ACTION_ADD                        = "add";
+    public static final String ACTION_UPDATE                     = "update";
+    public static final String ACTION_REMOVE                     = "remove";
+
+
+    /* message payload fields */
+    public static final String FIELD_ENABLED                     = "enabled";
+    public static final String FIELD_ADAPTERS                    = "adapters";
+    public static final String FIELD_ADAPTERS_TYPE               = "type";
+    public static final String FIELD_ADAPTERS_DEVICE_AUTHENTICATION_REQUIRED = "device-authentication-required";
+
+
+    private static final List<String> ACTIONS = Arrays.asList(ACTION_GET, ACTION_ADD, ACTION_UPDATE, ACTION_REMOVE);
+
+    /**
+     * The name of the Tenant API endpoint.
+     */
+    public static final String TENANT_ENDPOINT = "tenant";
+
+    /**
+     * The vert.x event bus address to which inbound registration messages are published.
+     */
+    public static final String EVENT_BUS_ADDRESS_TENANT_IN = "tenant.in";
+
+    /**
+     * Creates a JSON object from a Tenant API request message.
+     *
+     * @param message The AMQP 1.0 tenant request message.
+     * @return The tenant message created from the AMQP message.
+     * @throws NullPointerException if message is {@code null}.
+     * @throws DecodeException if the message contains a body that cannot be parsed into a JSON object.
+     */
+    public static JsonObject getTenantMsg(final Message message) {
+        Objects.requireNonNull(message);
+        final String subject = message.getSubject();
+        final String tenantId = MessageHelper.getTenantIdAnnotation(message);
```

Comment:
FMPOV we are reading the tenant ID from an application-property (not from an annotation), don't we?

Reply:
Good point, actually we have this "wrong" in all our APIs so far.
"Wrong", since so far there always is a filter active that propagates e.g. the tenantId from the resourceIdentifier to the message annotations, otherwise upper code would not even work.

Would consider this as an extra point to fix this in a followup (for **all** API implementations) and create a ticket for that, WDYT?